collect_app/src/main/java/org/odk/collect/android/widgets/GeoPointWidget.java

```diff
@@ -60,101 +55,71 @@
+        // Determine the accuracy threshold to use.
         String acc = prompt.getQuestion().getAdditionalAttribute(null, ACCURACY_THRESHOLD);
         if (acc != null && acc.length() != 0) {
             accuracyThreshold = Double.parseDouble(acc);
         } else {
             accuracyThreshold = DEFAULT_LOCATION_ACCURACY;
         }
 
-        // Determine whether or not to use the plain, maps, or mapsV2 activity
-        String appearance = prompt.getAppearanceHint();
-
-        // use mapsV2 if it is available and was requested;
-        useMapsV2 = useMapsV2(context);
-        if (appearance != null && appearance.toLowerCase(Locale.US).contains(WidgetAppearanceUtils.PLACEMENT_MAP) && useMapsV2) {
-            draggable = true;
-            useMaps = true;
-        } else if (appearance != null && appearance.toLowerCase(Locale.US).contains(WidgetAppearanceUtils.MAPS) && useMapsV2) {
-            draggable = false;
-            useMaps = true;
-        } else {
-            useMaps = false;
+        // Determine whether to use the map and whether the point should be draggable.
+        useMap = false;
+        if (MapConfigurator.getCurrentSource(context).isAvailable(context)) {
+            if (hasAppearance(prompt, PLACEMENT_MAP)) {
+                draggable = true;
+                useMap = true;
+            } else if (hasAppearance(prompt, MAPS)) {
+                draggable = false;
+                useMap = true;
+            }
         }
 
         readOnly = prompt.isReadOnly();
-
         answerDisplay = getCenteredAnswerTextView();
 
-        viewButton = getSimpleButton(getContext().getString(R.string.get_point), R.id.get_point);
-
         getLocationButton = getSimpleButton(R.id.get_location);
 
         // finish complex layout
-        // control what gets shown with setVisibility(View.GONE)
         LinearLayout answerLayout = new LinearLayout(getContext());
         answerLayout.setOrientation(LinearLayout.VERTICAL);
         answerLayout.addView(getLocationButton);
-        answerLayout.addView(viewButton);
         answerLayout.addView(answerDisplay);
         addAnswerView(answerLayout);
 
         // Set vars Label/text for button enable view or collect...
         boolean dataAvailable = false;
-        String s = prompt.getAnswerText();
-        if (s != null && !s.equals("")) {
+        String answer = prompt.getAnswerText();
+        if (answer != null && !answer.equals("")) {
             dataAvailable = true;
-            setBinaryData(s);
+            setBinaryData(answer);
         }
         updateButtonLabelsAndVisibility(dataAvailable);
-
     }
 
     private void updateButtonLabelsAndVisibility(boolean dataAvailable) {
-        // BUT for mapsV2, we only show the getLocationButton, altering its text.
-        // for maps, we show the view button.
-
-        if (useMapsV2 && useMaps) {
-            // hide the view button
-            viewButton.setVisibility(View.GONE);
+        if (useMap) {
```

Comment:
Yup—I removed the viewButton because it's never used.  `useMaps` is only true when `useMapsV2` is true, so  `viewButton.setVisibility(View.VISIBLE)` is never executed.